lib/Sema/CSGen.cpp

```diff
@@ -1312,10 +1311,9 @@ namespace {
       // Create an overload choice referencing this declaration and immediately
       // resolve it. This records the overload for use later.
       auto tv = CS.createTypeVariable(locator, TVO_CanBindToLValue);
-      CS.resolveOverload(locator, tv,
-                         OverloadChoice(Type(), E->getDecl(),
-                                        E->isSpecialized(), CS));
-      
+      CS.resolveOverload(locator, tv, OverloadChoice(Type(), E->getDecl(), CS,
+                                                     E->isSpecialized()));
+
```

Comment:
The type-checker part of this work is probably going to conflict really hard with Doug's current patch about removing function argument labels from unapplied calls.  It might make sense to just wait until that lands.